ethereum_bridge/src/protocol/transactions/ethereum_events/mod.rs

```diff
@@ -138,11 +137,16 @@ where
     H: 'static + StorageHasher + Sync,
 {
     let eth_msg_keys = vote_tallies::Keys::from(&update.body);
-
-    // we arbitrarily look at whether the seen key is present to
-    // determine if the /eth_msg already exists in storage, but maybe there
-    // is a less arbitrary way to do this
-    let exists_in_storage = wl_storage.has_key(&eth_msg_keys.seen())?;
+    let exists_in_storage = 'exists: {
+        let Some(seen) = votes::storage::maybe_read_seen(wl_storage, &eth_msg_keys)? else {
+            break 'exists false;
+        };
+        if seen {
+            tracing::debug!(?update, "Ethereum event is already seen");
+            return Ok((ChangedKeys::default(), false));
+        }
+        true
+    };
```

Comment:
```suggestion
    let exists_in_storage =  if let Some(seen) = votes::storage::maybe_read_seen(wl_storage, &eth_msg_keys)? {
        if seen {
            tracing::debug!(?update, "Ethereum event is already seen");
            return Ok((ChangedKeys::default(), false));
        }
        true
    } else {
        false
    };
```

Reply:
Let's not get overly excited about rust's new goto statements :stuck_out_tongue_closed_eyes: